in_gdummy.go

```diff
@@ -17,18 +21,85 @@ type gdummyPlugin struct {
 	counterSuccess metric.Counter
 	counterFailure metric.Counter
 	log            plugin.Logger
+	gen            dataGen
+	dummyMsg       map[string]string
+	template       *LogGenerator
+	rate           time.Duration
 }
 
+type dataGen uint
+
+const (
+	genDummy dataGen = iota
+	genTemplate
+)
+
 func (plug *gdummyPlugin) Init(ctx context.Context, fbit *plugin.Fluentbit) error {
 	plug.counterSuccess = fbit.Metrics.NewCounter("operation_succeeded_total", "Total number of succeeded operations", "gdummy")
 	plug.counterFailure = fbit.Metrics.NewCounter("operation_failed_total", "Total number of failed operations", "gdummy")
 	plug.log = fbit.Logger
 
+	if msg := fbit.Conf.String("dummy"); msg != "" {
+		if err := json.Unmarshal([]byte(msg), &plug.dummyMsg); err != nil {
+			return err
+		}
+	}
+
+	switch fbit.Conf.String("datagen") {
+	case "":
+		fallthrough
+	case "dummy":
+		plug.gen = genDummy
+		if plug.dummyMsg == nil {
+			return errors.New("invalid config: 'dummy' not set with datagen=dummy")
+		}
+	case "template":
+		plug.gen = genTemplate
+		fieldsStr := fbit.Conf.String("fields")
```

Comment:
I'm worried about field order.
Since Go plugins sends the records as `map[string]any` it won't keep the order of these fields. But the user might expect them in the given order.